osu.Game/Overlays/SkinEditor/SkinEditorChangeHandler.cs

```diff
@@ -33,7 +34,7 @@ public SkinEditorChangeHandler(Drawable targetScreen)
                 return;
 
             components = new BindableList<ISerialisableDrawable> { BindTarget = firstTarget.Components };
-            components.BindCollectionChanged((_, _) => SaveState());
+            components.BindCollectionChanged((_, _) => SaveState(), true);
```

Comment:
Just discovered this cannot be here, as it saves blank state when loading components for the first time.

Reply:
Okay, well, then I guess this part has to be reverted.

Reply:
Seems that putting the binding into `LoadComplete()` fixes it. I don't have enough knowledge about the framework to know if `SkinEditorChangeHandler.LoadComplete()` gets always called after `SkinBlueprintContainer.LoadComplete()`. In that case it should be fine.

Reply:
We usually put `BindValueChanged()` calls in `LoadComplete()` yes, but in this particular case it's likely that doing so would be nothing less than a crude timing hack. There's really no reason why the proper components shouldn't be present in the target screen at the point of ctor - and if there are issues with components not being present at that time, it should be the screens' responsibility to signal when it is valid to save the initial state rather than have the change handler attempt to guess based on its own lifetime.

Reply:
> Seems that putting the binding into LoadComplete() fixes it.

Don't worry about this, turns out I had some other changed that I was testing with, to not confuse you as in why it should work. Still gonna make separate PR.